pkg/function/utils.go

```diff
@@ -27,8 +28,57 @@ import (
 const (
 	// FunctionOutputVersion returns version
 	FunctionOutputVersion = "version"
+
+	KanisterPodCustomLabelsEnv      = "KANISTER_POD_CUSTOM_LABELS"
+	KanisterPodCustomAnnotationsEnv = "KANISTER_POD_CUSTOM_ANNOTATIONS"
 )
 
+func parseToLabelSelector(envKey string) (bool, *map[string]string) {
+	val, ok := os.LookupEnv(envKey)
+	if !ok || val == "" {
+		return false, nil
+	}
+	ls, err := metav1.ParseToLabelSelector(val)
+	if err != nil {
+		return false, nil
+	}
+	return true, &ls.MatchLabels
+}
+
+func getKanisterPodLabels() (bool, *map[string]string) {
+	return parseToLabelSelector(KanisterPodCustomLabelsEnv)
+}
+
+func getKanisterPodAnnotations() (bool, *map[string]string) {
+	return parseToLabelSelector(KanisterPodCustomAnnotationsEnv)
+}
+
+// SetLabelsToPodOptionsIfRequired sets labels to PodOptions
+func SetLabelsToPodOptionsIfRequired(options *kube.PodOptions) {
+	updateNeeded, labels := getKanisterPodLabels()
+	if updateNeeded {
```

Comment:
What are the cases when `updateNeeded` is false and `labels` is non-nil?

Reply:
`updatedNeedeed` would be false if user has not provided labels/annotations for kanister annotation while installing Kanister. And labels would be non-nil if the kube.options struct that we have initialised doesn't have labels/annotations already specified.

Reply:
Do we really need `updateNeeded`? Can we just not check if labels is nil?
```suggestion
	labels := getKanisterPodLabels()
	if labels != nil {
```